plugins/statistics.koplugin/main.lua

```diff
@@ -175,17 +176,49 @@ function ReaderStatistics:initData()
     -- Update these numbers to what's actually stored in the settings
     -- (not that "notes" is invalid and does not represent edited highlights)
     self.data.highlights, self.data.notes = self.ui.bookmark:getNumberOfHighlightsAndNotes()
-    self.curr_total_time = 0
-    self.curr_total_pages = 0
     self.id_curr_book = self:getIdBookDB()
     self.total_read_pages, self.total_read_time = self:getPageTimeTotalStats(self.id_curr_book)
     if self.total_read_pages > 0 then
         self.avg_time = self.total_read_time / self.total_read_pages
     else
-        self.avg_time = 0
+        -- NOTE: Possibly less weird-looking than initializing this to 0?
+        self.avg_time = math.floor(0.75 * self.page_max_read_sec)
+        logger.info("ReaderStatistics: Initializing average time per page at 75% of the max value, i.e.,", self.avg_time)
     end
 end
 
+-- Reset the (volatile) stats on page count changes (e.g., after a font size update)
+function ReaderStatistics:onUpdateToc()
+    local new_pagecount = self.view.document:getPageCount()
+
+    if new_pagecount ~= self.data.pages then
+        logger.info("ReaderStatistics: Pagecount change, clearing volatile book statistics")
+        -- Clear volatile stats for current book
+        self:resetVolatileStats()
```

Comment:
You reset/drop what you have (might be 30 pages and then I change font size) without saving them to DB ?

Reply:
I wondered about that, but if I flush to db instead, I flush potentially bogus data (c.f., the fixme below).

I don't think there's a perfect solution here :/.

Reply:
(The flush to db will call resetVolatileStats, which is why it's a potentially viable alternative, with a different set of drawbacks).

Reply:
Then, the 50 pages you flushed the last time before these new 30 ones are as bogus - or as valuable - as these 30 ones. No reason to make a distinction.

Reply:
Either you drop them all (and I don't want you to :) - or you/we live with mismatched (but complete/not dropped) set of data like we did - until we find a better idea (like suggested in the other PR, %ooooo of doc height (millionth of doc height) to rescale to page height when displaying stats - but it has some caveats, small pages, etc...)

Reply:
(For the record, we already display different data, like the nb of pages read I think, when viewing the stats for the currently opened book vs. viewing them for this book later once it is closed - not the same data is available - but everybody and me seem fine with that :) so, I wouldn't bother if the displayed info might have some mismatch - as long as the data in the DB is sane and display glitches can be fixed later.)

Reply:
As a compromise, in ec14c34, we flush, and in the rare cases `insertDB` wouldn't actually flush (very few pages read), we clear the volatile data.

This should hopefully cover the use-case where you open a new book, realize the font sizes are all out of whack, and tweak that before actually starting reading ;).